src/nodes.coffee

```diff
@@ -776,16 +840,30 @@ exports.Literal = class Literal extends Base
   compileNode: (o) ->
     [@makeCode @value]
 
+  astProps: ['value']
```

Comment:
Why do they need to be AST properties? Why can’t `value` simply be a property on the node, like any other property?

Reply:
@GeoffreyBooth see previous comment, think of `astProps` as similar to `children`, where it's providing a "pointer" to certain properties on the node
> Why can’t `value` simply be a property on the node, like any other property?

`value` is indeed a property on the node, this is just a declarative way of saying "my `value` property belongs on the AST as its `value` field"

Reply:
Why do we need `astProps` at all, instead of just using `children`? So that the AST node properties can differ from the class, e.g. to match Babel? Couldn’t we just rename various `children` properties as needed instead? Or perhaps be less concerned about matching Babel, at least for now?

What I’d like is to have the AST JSON output and the JS string output share as much code as possible. The existing node classes’ properties and methods aren’t sacred, we can modify them.